app/recurring_expenses/views.py

```diff
@@ -123,53 +124,89 @@ def create_recurring_expense(request):
+    )
+    writer = csv.writer(response)
+    writer.writerow(
+        [
+            "Active",
+            "Particulars",
+            "Amount",
+            "Currency",
+            "Amount NZD",
+            "Period (Months)",
+        ]
+    )
+    for expense in expenses:
+        amount_nzd = (1 / fx_rates[expense.currency]) * float(expense.amount)
+        expense.amount_nzd = amount_nzd
         writer.writerow(
             [
-                "Active",
-                "Particulars",
-                "Amount",
-                "Currency",
-                "Amount NZD",
-                "Period (Months)",
+                expense.active,
+                expense.particulars,
+                format(expense.amount, ".2f"),
+                expense.currency,
+                format(expense.amount_nzd, ".2f"),
+                expense.period,
             ]
         )
-        #
-        # Get expenses and amount in NZD
-        # Copied from class IndexView, maybe this should be a function?
-        # Exporting each expense to csv within this though so we don't
-        # have to run the for loop twice
-        #
-        fx_rates = get_fxRate_nzd()
-        expenses = RecurringExpense.objects.all()
-        for expense in expenses:
-            amount_nzd = (1 / fx_rates[expense.currency]) * float(expense.amount)
-            expense.amount_nzd = amount_nzd
-            writer.writerow(
-                [
-                    expense.active,
-                    expense.particulars,
-                    format(expense.amount, ".2f"),
-                    expense.currency,
-                    format(expense.amount_nzd, ".2f"),
-                    expense.period,
-                ]
-            )
-    else:
-        response = HttpResponse(status=204)
+    return response
+
 
+def export_to_json(expenses):
+    data = []
+    fx_rates = get_fxRate_nzd()
+    for expense in expenses:
+        amount_nzd = (1 / fx_rates[expense.currency]) * float(expense.amount)
+        expense.amount_nzd = amount_nzd
+        data.append({
+            "Active": expense.active,
+            "Particulars": expense.particulars,
+            "Amount": format(expense.amount, ".2f"),
+            "Currency": expense.currency,
+            "Amount NZD": format(expense.amount_nzd, ".2f"),
+            "Period (Months)": expense.period,
+        })
+
+    json_data = json.dumps(data, indent=4)
+
+    response = HttpResponse(
+        content_type="application/json",
+        headers={
+            "Content-Disposition": 'attachment; filename="recurring_expenses.json"'
+        },
+    )
+    response.write(json_data)
     return response
 
 
+@login_required
+def export_recurring_expenses(request, export_format):
+    # Assuring that the url parameter is not case-sensitive.
+    export_format = export_format.lower()
+
+    if export_format not in _EXPORT_FORMATS:
+        return HttpResponse(status=204)
```

Comment:
I don't understand why this works but it works and I like it. I wonder if it would also work with a HTTP 400 response code?

Something I would be interested to explore in the future. Nice work